libraries/AP_ICEngine/AP_ICEngine.cpp

```diff
@@ -357,7 +357,8 @@ void AP_ICEngine::update(void)
             allow_single_start_while_disarmed = false;
         } else {
             // check if we are blocking disarmed starts
-            if (!allow_single_start_while_disarmed) {
+            if ((hal.util->safety_switch_state() == AP_HAL::Util::SAFETY_DISARMED) ||
```

Comment:
Won't this still be a issue if `NO_RUNNING_WHILE_DISARMED` is not set?

Reply:
I mean yes, but presumably you at least requested that? I think it's still safer then otherwise.